src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -5521,75 +5542,87 @@ private string CreateManifestString()
             // Write out the channels
             if (channelTab != null)
             {
-                sb.AppendLine(" <channels>");
-                var sortedChannels = new List<KeyValuePair<int, ChannelInfo>>();
-                foreach (KeyValuePair<int, ChannelInfo> p in channelTab) { sortedChannels.Add(p); }
-                sortedChannels.Sort((p1, p2) => -Comparer<ulong>.Default.Compare(p1.Value.Keywords, p2.Value.Keywords));
-                foreach (KeyValuePair<int, ChannelInfo> kvpair in sortedChannels)
+                IEnumerable<KeyValuePair<int, ChannelInfo>> channels = channelTab;
+                sb?.AppendLine(" <channels>");
+                if (WillBuild)
+                {
+                    var sortedChannels = new List<KeyValuePair<int, ChannelInfo>>();
```

Comment:
I am curious why you changed it to only do the sorting if the manifest will be emitted? I don't think it will cause problems but I also don't see the advantage

Reply:
In retrospect, I realize how unnecessary omitting the sorting is. Both the channels and opcodes appeared to never even be generated.

The stringTab seems like it could benefit the most. However, it contains only 62 entries or so. The benefit of not sorting them appears to be negligible at best and potentially could result in more headaches down the road. I’ll remove those changes.